samples/bluetooth/hci_uart/src/main.c

```diff
@@ -311,46 +297,12 @@ void main(void)
 	/* Enable the raw interface, this will in turn open the HCI driver */
 	bt_enable_raw(&rx_queue);
 
-	if (IS_ENABLED(CONFIG_BT_WAIT_NOP)) {
-		/* Issue a Command Complete with NOP */
-		int i;
-
-		const struct {
-			const uint8_t h4;
-			const struct bt_hci_evt_hdr hdr;
-			const struct bt_hci_evt_cmd_complete cc;
-		} __packed cc_evt = {
-			.h4 = H4_EVT,
-			.hdr = {
-				.evt = BT_HCI_EVT_CMD_COMPLETE,
-				.len = sizeof(struct bt_hci_evt_cmd_complete),
-			},
-			.cc = {
-				.ncmd = 1,
-				.opcode = sys_cpu_to_le16(BT_OP_NOP),
-			},
-		};
-
-		for (i = 0; i < sizeof(cc_evt); i++) {
-			uart_poll_out(hci_uart_dev,
-				      *(((const uint8_t *)&cc_evt)+i));
-		}
-	}
-
-	/* Spawn the TX thread and start feeding commands and data to the
-	 * controller
-	 */
-	k_thread_create(&tx_thread_data, tx_thread_stack,
-			K_THREAD_STACK_SIZEOF(tx_thread_stack), tx_thread,
-			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
+	err =  h4_uart_init(&transport, hci_uart_dev, &config);
+	__ASSERT(err >= 0, "Unexpected error: %d", err);
 
 	while (1) {
-		struct net_buf *buf;
-
-		buf = net_buf_get(&rx_queue, K_FOREVER);
-		err = h4_send(buf);
-		if (err) {
-			LOG_ERR("Failed to send");
-		}
+		err = h4_uart_write(&transport,
```

Comment:
I think you are fixing:

https://github.com/zephyrproject-rtos/zephyr/issues/2955

Reply:
yes